tests/e2e-demos/e2e-demo.go

```diff
@@ -199,7 +199,10 @@ var _ = framework.E2ESuiteDescribe(Label("e2e-demo"), func() {
 					Expect(err).ShouldNot(HaveOccurred())
 
 					Eventually(func() bool {
-						return fw.IntegrationController.HaveHACBSTestsSucceeded(snapshot)
+						if snapshot != nil {
+							return fw.IntegrationController.HaveHACBSTestsSucceeded(snapshot)
```

Comment:
IMO this won't help, because `if snapshot == nil`, it will keep returning `false` in the `Eventually` until it times out.
We'd like to try to get the application snapshot again and perform this check again.

Also wouldn't it make sense to update the [GetApplicationSnapshot method](https://github.com/redhat-appstudio/e2e-tests/blob/c1f0f6ffeae1edc8cc07b6afd8a42249160f861b/pkg/utils/integration/controller.go#L58)?
It doesn't return an error [in case the snapshot is not found](https://github.com/redhat-appstudio/e2e-tests/blob/c1f0f6ffeae1edc8cc07b6afd8a42249160f861b/pkg/utils/integration/controller.go#L80), which might be misleading

Reply:
My bad, sorry. You are right. Thank you, @psturc!

Regarding the fact that `GetApplicationSnapshot` does not return an error in case the snapshot is not found, I agree with you that it is misleading. However, in this case, I think the error check will need to be removed (in case we change it to return an error) to not "break" the `It` in case the snapshot is not yet found

Reply:
Sure! I will also update the other methods like`GetComponent` that present the same situation.